src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -272,15 +273,58 @@ CHIP_ERROR DiscoveryImplPlatform::Advertise(const OperationalAdvertisingParamete
     mOperationalAdvertisingParams = params;
     // TODO: There may be multilple device/fabrid ids after multi-admin.
 
+    TextEntry crmpRetryIntervalEntries[2];
+    size_t textEntrySize = 0;
+    char crmpRetryIntervalIdleBuf[8];
+    char crmpRetryIntervalActiveBuf[8];
+    uint32_t crmpRetryIntervalIdle, crmpRetryIntervalActive;
+    params.GetCRMPRetryIntervals(crmpRetryIntervalIdle, crmpRetryIntervalActive);
+
+    // TODO: Issue #5833 - CRMP retry intervals should be updated on the poll period value
+    // change or device type change.
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    if (chip::DeviceLayer::ConnectivityMgr().GetThreadDeviceType() ==
+        chip::DeviceLayer::ConnectivityManager::kThreadDeviceType_SleepyEndDevice)
+    {
+        uint32_t sedPollPeriod;
+        ReturnErrorOnFailure(chip::DeviceLayer::ThreadStackMgr().GetPollPeriod(sedPollPeriod));
+        // Increment default CRMP retry intervals by SED poll period to be on the safe side
+        // and avoid unnecessary retransmissions.
+        crmpRetryIntervalIdle += sedPollPeriod;
+        crmpRetryIntervalActive += sedPollPeriod;
+    }
+#endif
+
+    if (crmpRetryIntervalIdle > 3600000)
+    {
+        ChipLogProgress(Discovery, "CRMP retry interval idle value exceeds allowed range od 1 hour, using maximal available",
+                        chip::ErrorStr(error));
+        crmpRetryIntervalIdle = 3600000;
+    }
+    snprintf(crmpRetryIntervalIdleBuf, sizeof(crmpRetryIntervalIdleBuf), "%u", crmpRetryIntervalIdle);
```

Comment:
nit: should we support skipping the entry in case it's equal to 0 or something like that? Just thinking.

Reply:
Hmm, I didn't see in the spec that it is not allowed to set those values to 0.